### PR TITLE
refactor: expose warehouse utils in common

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #9711 

### Description:

Creates `common/utils/warehouse.ts` and adds `getFieldQuoteChar`. 
Now this util is used across the FE and BE

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by rephus — APPROVED** [review]
Yep, that's what I had in mind , thanks

File: packages/common/src/utils/warehouse.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { WarehouseTypes } from '../types/projects';
+import assertUnreachable from './assertUnreachable';
+
+export const getFieldQuoteChar = (
+    warehouseType: WarehouseTypes | undefined,
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/common/src/utils/warehouse.ts:5]: Yeah, I think that's fair. Another option is not remove the undefined, and do `warehouse.type !== undefined ?  getFieldQuoteChar(warehouse.type) : '"' ` on the frontend

**Review by rephus — APPROVED** [review]

**Comment by joaoviana** [comment]
looking at tests

File: packages/backend/src/queryBuilder.mock.ts [hunk 2]
```diff
@@ -50,7 +49,9 @@ export const warehouseClientMock: WarehouseClient = {
 };
 
 export const bigqueryClientMock: WarehouseClient = {
-    credentials: {} as CreateWarehouseCredentials,
+    credentials: {
+        type: WarehouseTypes.BIGQUERY,
```
Comment (joaoviana) [inline packages/backend/src/queryBuilder.mock.ts:53]: had a missing `type`

**Pull request merged**